runtime/lua/vim/lsp/client.lua

```diff
@@ -210,7 +207,7 @@ local validate = vim.validate
 --- Map with language server specific settings. These are returned to the
 --- language server if requested via `workspace/configuration`. Keys are
 --- case-sensitive.
---- @field settings table
+--- @field settings lsp.LSPObject
```

Comment:
```suggestion
--- @field settings lsp.LspObject
```

Reply:
`lsp.LspObject` isn't defined. Everything in `lsp.*` is auto generated from the protocol.